internal/promotion/runner/builtin/git_pr_opener.go

```diff
@@ -159,6 +164,7 @@ func (g *gitPROpener) run(
 			Status: kargoapi.PromotionPhaseSucceeded,
 			Output: map[string]any{
 				stateKeyPRNumber: pr.Number,
+				stateKeyRepoURL:  cfg.RepoURL,
```

Comment:
May I suggest that we use this opportunity to stop calling it a "pr number" and start calling it an ID? I.e., I feel this might be slightly easier to use:

```yaml
output:
  pr:
    id: 1234
    url: https://github.com/org/example/pull/1234
```

Reply:
Good thought.

It is a breaking change, so @fykaa docs will need to be updated accordingly. `getPRNumber()` will also need to be modified to look at `pr.id` first and then fall back on `prNumber`, _unless_ we decide to just do away with the [short-circuiting](https://github.com/akuity/kargo/pull/3884#discussion_r2049506582) this step does, in which case, `getPRNumber()` is code we can jettison.

@hiddeco any thoughts on that?

Reply:
Although... this is a somewhat nasty breaking change. Anyone _using_ that output in an expression would have to rewrite expressions to look for the new output structure and then fall back on the old one. That seems incredibly onerous.

We could either deprecate without ever removing or live with the way things are. 

@hiddeco?

Reply:
My idea was to maintain the `prNumber` field for some time, and remove it at a later moment.